flink-core/src/main/java/org/apache/flink/metrics/MetricRegistry.java

```diff
@@ -75,77 +78,92 @@ public MetricRegistry(Configuration config) {
 		this.delimiter = delim;
 
 		// second, instantiate any custom configured reporters
-		
-		final String className = config.getString(ConfigConstants.METRICS_REPORTER_CLASS, null);
-		if (className == null) {
+		this.reporters = new ArrayList<>();
+
+		final String definedReporters = config.getString(ConfigConstants.METRICS_REPORTERS_LIST, null);
+
+		if (definedReporters == null) {
+			// no reporters defined
 			// by default, don't report anything
 			LOG.info("No metrics reporter configured, no metrics will be exposed/reported.");
-			this.reporter = null;
 			this.executor = null;
-		}
-		else {
-			MetricReporter reporter;
-			ScheduledExecutorService executor = null;
-			try {
-				String configuredPeriod = config.getString(ConfigConstants.METRICS_REPORTER_INTERVAL, null);
-				TimeUnit timeunit = TimeUnit.SECONDS;
-				long period = 10;
-				
-				if (configuredPeriod != null) {
-					try {
-						String[] interval = configuredPeriod.split(" ");
-						period = Long.parseLong(interval[0]);
-						timeunit = TimeUnit.valueOf(interval[1]);
+		} else {
+			// we have some reporters so
+			String[] namedReporters = definedReporters.split(",");
+			for (String namedReporter : namedReporters) {
```

Comment:
yes. but this will go completely unnoticed to a user since nothing is being logged.